VMEncryption/main/DiskUtil.py

```diff
@@ -1091,7 +1093,24 @@ def is_os_disk_lvm(self):
 
         return DiskUtil.os_disk_lvm
 
-    def should_skip_for_inplace_encryption(self, device_item, encrypt_volume_type):
+    def is_data_disk(self, device_item, azure_devices):
+        # Root disk
+        if device_item.device_id.startswith('00000000-0000'):
+            self.logger.log(msg="skipping root disk", level=CommonVariables.WarningLevel)
+            return False
+        # Resource Disk. Not considered a "data disk" exactly (is not attached via portal and we have a separate code path for encyrpting it)
+        if device_item.device_id.startswith('00000000-0001'):
+            self.logger.log(msg="skipping resource disk", level=CommonVariables.WarningLevel)
+            return False
+
+        for azure_blk_item in azure_devices:
+            if azure_blk_item.name == device_item.name:
+                self.logger.log(msg="the mountpoint is the azure disk root or resource, so skip it.")
+                return False
```

Comment:
why are there 2 distinct ways to check for root disk and resource disks?

Reply:
Just to safe guard against one of the methods not working in Gen 2.
Doesn't hurt to have two ways even for Gen 1.